### ISSUE TITLE
[BUG] POST with System.Object type results in InvalidOperationException: Should not get here.

### ISSUE THREAD

**Issue by acidshards** [reactiveui/refit#1079]
**Describe the bug**
When executing a POST with an object as parameter Refit v6 will return an InvalidOperationException: Should not get here.

**Steps To Reproduce**
1. 
```csharp
  public interface IRefitContract
    {
        [Post("/api/users")]
        Task Post(Person person);
    }

    public class Person
    {
        public int name { get; set; }
        public object job { get; set; }
    }
```
2. Results in
```
System.InvalidOperationException: Should not get here.
   at Refit.ObjectToInferredTypesConverter.Write(Utf8JsonWriter writer, Object objectToWrite, JsonSerializerOptions options) in /_/Refit/SystemTextJsonContentSerializer.cs:line 127
   at System.Text.Json.Serialization.JsonConverter`1.TryWrite(Utf8JsonWriter writer, T& value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.JsonPropertyInfo`1.GetMemberAndWriteJson(Object obj, WriteStack& state, Utf8JsonWriter writer)
   at System.Text.Json.Serialization.Converters.ObjectDefaultConverter`1.OnTryWrite(Utf8JsonWriter writer, T value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.Serialization.JsonConverter`1.TryWrite(Utf8JsonWriter writer, T& value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.Serialization.JsonConverter`1.WriteCore(Utf8JsonWriter writer, T& value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.Serialization.JsonConverter`1.WriteCoreAsObject(Utf8JsonWriter writer, Object value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.JsonSerializer.WriteCore[TValue](JsonConverter jsonConverter, Utf8JsonWriter writer, TValue& value, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.JsonSerializer.WriteAsyncCore[TValue](Stream utf8Json, TValue value, Type inputType, JsonSerializerOptions options, CancellationToken cancellationToken)
   at System.Net.Http.Json.JsonContent.SerializeToStreamAsyncCore(Stream targetStream, Boolean async, CancellationToken cancellationToken)
   at System.Net.Http.HttpContent.CopyToAsyncCore(ValueTask copyTask)
   at Refit.RequestBuilderImplementation.<>c__DisplayClass18_2.<<BuildRequestFactoryForMethod>b__3>d.MoveNext() in /_/Refit/RequestBuilderImplementation.cs:line 583
--- End of stack trace from previous location where exception was thrown ---
   at System.Net.Http.PushStreamContent.SerializeToStreamAsync(Stream stream, TransportContext context) in /_/Refit/PushStreamContent.cs:line 138
   at System.Net.Http.HttpContent.CopyToAsyncCore(ValueTask copyTask)
   at System.Net.Http.HttpConnection.SendRequestContentAsync(HttpRequestMessage request, HttpContentWriteStream stream, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, CancellationToken cancellationToken)
```

**Expected behavior**
Successful POST

**Environment**

- OS: Windows 10
- Version:   6.0.15
- Working Version: 5.2.4

**Comment by clairernovotny** [comment]
The System.Text.Json serializer doesn't support writing `object` directly.

**Comment by acidshards** [comment]
Thanks for the information, it's all clear now!

**Comment by clairernovotny** [comment]
Fixed in #1088
